joeynmt/prediction.py

```diff
@@ -117,15 +125,17 @@ def validate_on_data(model: Model, data: Dataset,
             # run as during inference to produce translations
             output, attention_scores = run_batch(
                 model=model, batch=batch, beam_size=beam_size,
-                beam_alpha=beam_alpha, max_output_length=max_output_length)
+                beam_alpha=beam_alpha, max_output_length=max_output_length,
+                n_best=n_best)
 
             # sort outputs back to original order
-            all_outputs.extend(output[sort_reverse_index])
-            valid_attention_scores.extend(
-                attention_scores[sort_reverse_index]
-                if attention_scores is not None else [])
+            for reverse_index in sort_reverse_index:
+                all_outputs.append(output[reverse_index])
+                valid_attention_scores.append(
+                    attention_scores[reverse_index]
+                    if attention_scores is not None else [])
 
-        assert len(all_outputs) == len(data)
+            assert len(all_outputs) == len(data)
```

Comment:
This indentation seems too much. Is it intended?

Reply:
I've made the assertion to be checked in each loop. You're right, I can use the previous indent and run it just once.